src/globus_sdk/services/transfer/client.py

```diff
@@ -411,7 +412,10 @@ def endpoint_search(
             endpoint. May cause BadRequest or PermissionDenied errors if the endpoint ID
             given is not valid for this operation.
         :param filter_non_functional: Limit search to endpoints which have the
-            'non_functional' flag set to True or False.
+            'non_functional' flag set to True or False. Mutually exclusive with
+            ``filter_entity_type``.
+        :param filter_entity_type: Limit search to endpoints or collections by their
+            entity type. Mutually exclusive with ``filter_non_functional``.
```

Comment:
I have a question about this, which I realized when suggesting the Literal annotation above and don't see clarified here or in the Transfer docs. Are users limited to passing exactly one entity type here, or can they make any of these requests to combine them?
```
/endpoint_search?filter_entity_type=GCSv4_host/GCSv5_endpoint
/endpoint_search?filter_entity_type=GCSv4_host,GCSv5_endpoint
/endpoint_search?filter_entity_type=GCSv4_host&filter_entity_type=GCSv5_endpoint
```

If any of these are possible, I think it's best to handle an iterable of those `Literal` values, allowing for `filter_entity_type="GCSv5_endpoint"` as well as `filter_entity_type=["GCSv5_endpoint"]` and `filter_entity_type=("GCSv5_endpoint", "GCSv4_host")`.
It would make the annotations quite tricky but I'm happy to help if that reflects reality.

If these usages are not possible I count that as a reason in favor of the `Literal` annotation -- it clarifies that you can only pass one.

Either way, I think we should make a small tweak to clarify the Transfer API docs. I'll make a suggestion over there once I have clarity on what is/is not supported.

Reply:
@sirosen asked:

> I have a question about this, which I realized when suggesting the Literal annotation above and don't see clarified here or in the Transfer docs. Are users limited to passing exactly one entity type here, or can they make any of these requests to combine them?
> 
> ```
> /endpoint_search?filter_entity_type=GCSv4_host/GCSv5_endpoint
> /endpoint_search?filter_entity_type=GCSv4_host,GCSv5_endpoint
> /endpoint_search?filter_entity_type=GCSv4_host&filter_entity_type=GCSv5_endpoint
> ```
> 
> If any of these are possible, I think it's best to handle an iterable of those `Literal` values, allowing for `filter_entity_type="GCSv5_endpoint"` as well as `filter_entity_type=["GCSv5_endpoint"]` and `filter_entity_type=("GCSv5_endpoint", "GCSv4_host")`. It would make the annotations quite tricky but I'm happy to help if that reflects reality.
> 
> If these usages are not possible I count that as a reason in favor of the `Literal` annotation -- it clarifies that you can only pass one.
> 
> Either way, I think we should make a small tweak to clarify the Transfer API docs. I'll make a suggestion over there once I have clarity on what is/is not supported.

The value of `filter_entity_type`, if specified, must be one of the five recognized entity types. Combinations of entity types are not supported.

The Transfer API operation is a thin wrapper around a lower level function.

https://github.com/globusonline/koa/blob/ede225ab1e22b7661d242b9b04437f61f314becd/tapi/globus_org_rest/view/endpoint_search.py#L30-L41

That lower level function just checks for whether it is specified and in a set list of values.

https://github.com/globusonline/koa/blob/ede225ab1e22b7661d242b9b04437f61f314becd/pykoa/api/endpoint_search.py#L124-L132

Reply:
Implemented in commit 55cad902d0ddd8b929cd05425be14ac33e57cd66.